core/txpool/legacypool/legacypool.go

```diff
@@ -1808,8 +1808,9 @@ func (t *lookup) addAuthorities(tx *types.Transaction) {
 
 // removeAuthorities stops tracking the supplied tx in relation to its
 // authorities.
-func (t *lookup) removeAuthorities(hash common.Hash) {
-	for addr := range t.auths {
+func (t *lookup) removeAuthorities(tx *types.Transaction) {
+	hash := tx.Hash()
+	for _, addr := range tx.SetCodeAuthorities() {
```

Comment:
Just to mention that duplicated authorities could be returned. 

The txhash will be removed by the auth in the first occurrence, and not-found will occur for the second occurrence.

Reply:
fixed with [bcb9836](https://github.com/ethereum/go-ethereum/pull/31249/commits/bcb983693e93131a3b3489dc71cdd3771d4f3721)